foo() call result is 42